runtime/event_loop.cpp

```diff
@@ -59,32 +59,42 @@ bool EventLoop::run_event_loop(api::Engine *engine, double total_compute) {
   queue.get().event_loop_running = true;
   JSContext *cx = engine->cx();
 
-  while (true) {
+  do {
     // Run a microtask checkpoint
     js::RunJobs(cx);
 
     if (JS_IsExceptionPending(cx)) {
       exit_event_loop();
       return false;
     }
-    // if there is no interest in the event loop at all, just run one tick
-    if (interest_complete()) {
-      exit_event_loop();
-      return true;
-    }
 
     const auto tasks = &queue.get().tasks;
     size_t tasks_size = tasks->size();
+
     if (tasks_size == 0) {
+      if (interest_complete()) {
+        break;
+      }
       exit_event_loop();
```

Comment:
I think we still want to exit the event loop here?

```suggestion
      exit_event_loop();
      if (interest_complete()) {
        break;
      }
```

Reply:
There's already an `exit_event_loop` on the break path, so this isn't needed.

Reply:
Ooh, right—apologies!